zookeeper-server/zookeeper-server-3.5.6/src/main/java/com/yahoo/vespa/zookeeper/Reconfigurer.java

```diff
@@ -86,7 +88,26 @@ private void reconfigure(ZookeeperServerConfig newConfig) {
         joiningServers = joiningServers.isEmpty() ? null : joiningServers;
         log.log(Level.INFO, "Will reconfigure ZooKeeper cluster. Joining servers: " + joiningServers +
                             ", leaving servers: " + leavingServers);
-        zooKeeperReconfigure(connectionSpec(activeConfig), joiningServers, leavingServers);
+
+        String connectionSpec = connectionSpec(activeConfig);
+        boolean reconfigured = false;
+        Instant end = Instant.now().plus(retryReconfigurationPeriod);
+        // Loop reconfiguring since we might need to wait until another reconfiguration is finished before we can succeed
+        while ( ! reconfigured && Instant.now().isBefore(end)) {
+            try {
+                zooKeeperReconfigure(connectionSpec, joiningServers, leavingServers);
+                reconfigured = true;
+            } catch (KeeperException e) {
+                if ( ! (e instanceof KeeperException.ReconfigInProgress))
+                    throw new RuntimeException(e);
+                log.log(Level.INFO, "Will retry in 1 second");
+                try {
+                    Thread.sleep(1000);
```

Comment:
Consider making the sleeper configurable by having an instance of `Consumer<Duration>` in this class, then the unit test doesn't actually have to sleep.

Reply:
Consider extracting a constant for the sleep period that is used both in the log message and sleep parameter.

Reply:
Good suggestions, will implement